File: examples/src/Presentations/Properties/AddCustomDocumentProperties.cs
using System
import aspose.slides as slides

/*
This project uses Automatic Package Restore feature of NuGet to resolve Aspose.Slides for .NET API reference 
when the project is build. Please check https:#docs.nuget.org/consume/nuget-faq for more information. 
If you do not wish to use NuGet, you can manually download Aspose.Slides for .NET API from http:#www.aspose.com/downloads, 
install it and then add its reference to this project. For any issues, questions or suggestions 
please feel free to contact us using http:#www.aspose.com/community/forums/default.aspx
*/

namespace Aspose.slides.Examples.CSharp.Presentations.properties
{
    class AddCustomDocumentProperties
    {
        public static void Run()
        {
            #ExStart:AddCustomDocumentProperties
            # The path to the documents directory.
            dataDir = RunExamples.GetDataDir_PresentationProperties()

            # Instantiate the Presentation class
            with slides.Presentation() as presentation:

            # Getting Document Properties
            IDocumentProperties documentProperties = presentation.DocumentProperties

            # Adding Custom properties
            documentProperties["New Custom"] = 12
            documentProperties["My Name"] = "Mudassir"
            documentProperties["Custom"] = 124

            # Getting property name at particular index
            getPropertyName = documentProperties.GetCustomPropertyName(2)

            # Removing selected property
            documentProperties.RemoveCustomProperty(getPropertyName)

            # Saving presentation
            presentation.save(dataDir + "CustomDocumentProperties_out.pptx", slides.export.SaveFormat.PPTX)
            #ExEnd:AddCustomDocumentProperties
        }
    }
}

File: examples/src/presentations/properties/add_custom_document_properties.py
import aspose.slides as slides


def props_add_custom_document_properties(global_opts):
    # Instantiate the Presentation class
    with slides.Presentation() as presentation:
        # Getting Document Properties
        document_properties = presentation.document_properties

        # Adding Custom properties
        document_properties.set_custom_property_value("New Custom", 12)
        document_properties.set_custom_property_value("My Nam", "Mudassir")
        document_properties.set_custom_property_value("Custom", 124)

        # Getting property name at particular index
        property_name = document_properties.get_custom_property_name(2)

        # Removing selected property
        document_properties.remove_custom_property(property_name)

        # Saving presentation
        presentation.save(global_opts.out_dir + "props_add_custom_document_properties_out.pptx",
                          slides.export.SaveFormat.PPTX)
